src/Servers/Kestrel/tools/CodeGenerator/TransportConnectionFeatureCollection.cs

```diff
@@ -27,6 +28,7 @@ public static string GenerateFile()
                 "IConnectionIdFeature",
                 "IConnectionTransportFeature",
                 "IConnectionItemsFeature",
+                "IPersistentStateFeature",
```

Comment:
I think we should handle this more like the IConnectionSocketFeature, and implement this for QuicStreamContext only, since we don't pool SocketConnections.

Reply:
We don't pool SocketConnections, but it is a consistent place to put state on HTTP/1.1, HTTP/2, HTTP/3 requests and have it preserved between requests.

Reply:
ping @halter73

Reply:
I'd rather this not be implemented by transports it's not supported on. We should implement this on Http1Connection, not SocketConnection, similarly to how you implement it in Http2Stream.

Reply:
Done